test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDropDownListExtensionsTest.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.TestCommon;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Core
+{
+    public class HtmlHelperDropDownListExtensionsTest
```

Comment:
In both of these classes, need to add more complex `SelectListItem`s, especially those that have a null `Value`, are `Disabled` or `Selected`, or are grouped. Note that `Selected` is used only when the helper is given a `null` expression.